### PR TITLE
Add oauth 1 to implement 1

### PR DESCRIPTION
**Pull request by pinterid**
**Checklist:**

 
 - [x] Have you added an explanation of what your changes do and why you'd like them to be included?
 - [x] Have you updated or added documentation for the change?
 - [x] Have you tested your changes with successful results?

**Type of Changes:**

 
 - [ ] Bug fix (non-breaking change which fixes an issue)
 - [x] New feature (non-breaking change which adds functionality)
 - [ ] Breaking change (fix or feature that would cause existing functionality to change)

**What is the current behavior? (link to any open issues here)**
- Currently the oauth GitHub provider is not provided from intel. #37 

**What is the new behavior (if this is a feature change)?**
- Now the oauth GitHub provider is provided from intel. #37

### CODE REVIEW

File: src/toolbox/GuidGenerator.ts [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+}
+//#endregion
+
+//#region > Export
+export default GuidGenerator;
+//#endregion
+
+/**
+ * SPDX-License-Identifier: (EUPL-1.2)
+ * Copyright © Simon Prast
```
Comment (schettn) [inline src/toolbox/GuidGenerator.ts:26]: This copyright tag is outdated.
Please change it to:  "Copyright © 2019-2020 Simon Prast"

File: src/utils/oauth/GithubProvider.ts [hunk 2]
```diff
@@ -0,0 +1,132 @@
...
+};
+//#endregion
+
+//#region > Exports
+export default GithubProvider;
+//#endregion
+
+/**
+ * SPDX-License-Identifier: (EUPL-1.2)
+ * Copyright © Simon Prast
```
Comment (schettn) [inline src/utils/oauth/GithubProvider.ts:131]: This copyright tag is outdated.
Please change it to:  "Copyright © 2019-2020 Simon Prast"

File: src/utils/oauth/index.ts [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+//#endregion
+
+//#region > Exports
+export default RSA;
+export { GithubProvider };
+//#endregion
+
+/**
+ * SPDX-License-Identifier: (EUPL-1.2)
+ * Copyright © Simon Prast
```
Comment (schettn) [inline src/utils/oauth/index.ts:17]: This copyright tag is outdated.
Please change it to:  "Copyright © 2019-2020 Simon Prast"

File: src/toolbox/GuidGenerator.ts [hunk 4]
```diff
@@ -0,0 +1,27 @@
+//#region > Functions
+/**
+ * @function Generate random guid string.
+ * @returns {string} A guid string.
+ * @description This function generates a random guid string.
+ */
+function GuidGenerator() {
+  let d = new Date().getTime();
+
+  return "xxxxxxxx-xxxx-4xxx-yxxx-xxxxxxxxxxxx".replace(/[xy]/g, function(c) {
```
Comment (schettn) [inline src/toolbox/GuidGenerator.ts:10]: Change this to a ES6 arrow function.
Ref: https://www.sitepoint.com/es6-arrow-functions-new-fat-concise-syntax-javascript/

File: src/utils/oauth/config.json [hunk 5]
```diff
@@ -0,0 +1,27 @@
+{
+  "providers": {
+    "github": {
+      "details": {
+        "dev": {
```
Comment (schettn) [inline src/utils/oauth/config.json:5]: Add documentation.
Example: /** @see {@link LINK_TO_ORGANIZATION |Organization} for further information. */

File: src/utils/oauth/config.json [hunk 6]
```diff
@@ -0,0 +1,27 @@
...
+          "clientId": "1440dd4c1d1c4c0fa124",
+          "clientSecret": "0723a2b5bfef27efc8b2d26d837ead239fa0b0e6",
+          "redirectUrl": "http://localhost:3000/redirect",
+          "scope": "repo, user:email, read:user, read:org"
+        },
+        "deploy": {
```
Comment (schettn) [inline src/utils/oauth/config.json:11]: Add documentation.
Example: /** @see {@link LINK_TO_ORGANIZATION |Organization} for further information. */

File: src/utils/oauth/config.json (hunk 5 above)
Comment (pinterid) [reply src/utils/oauth/config.json:5]: Comments are not permitted in config.json!

File: src/utils/oauth/config.json (hunk 6 above)
Comment (pinterid) [reply src/utils/oauth/config.json:11]: Comments are not permitted in config.json!

File: src/utils/oauth/GithubProvider.ts [hunk 7]
```diff
@@ -0,0 +1,132 @@
...
+// Contains the values for the provider
+import Config from "./config.json";
+//#endregion
+
+//#region > Providers
+// Set the default values needed for an OAuth-Request
+const state = GuidGenerator();
+let providerConfig = Config.providers.github;
+let details = providerConfig.details.deploy;
+
```
Comment (kleberbaum) [inline src/utils/oauth/GithubProvider.ts:18]: Please remove newline.

File: src/utils/oauth/index.ts [hunk 8]
```diff
@@ -0,0 +1,18 @@
+//#region > Imports
+//> RSA 
+// Contains authentication service
+import RSA from "react-very-simple-oauth";
```
Comment (kleberbaum) [inline src/utils/oauth/index.ts:4]: Please add a newline.

**Pull request merged**